book/src/developers/designs/replication.md

```diff
@@ -0,0 +1,465 @@
+
+In this example, if A were supplying to B, then A would supply `A {T9, T10}` and `C {T9 -> T13}`.
+This allows the replication to avoid full connection (where every node must contact every other
+node).
+
+In order to select the entries for supply, the database maintains an index of entries that are
+affected by any change for any cid. This allows range requests to be made for efficient selection of
+what entries were affected in any cid.
+
+After an incremental replication is applied, the RUV is updated to reflect the application of these
+differences.
+
+## Lagging / Advanced Consumers
+
+Replication relies on each node periodically communicating for incremental updates. This is because
+of _deletes_. A delete event occurs by a Live entry becoming a Tombstone. A tombstone is replicated
+over the live entry. Tomestones are then _reaped_ by each node individually once the replication
+delay window has passed.
+
+This delay window is there to allow every node the chance to have the tombstone replicated to it, so
+that all nodes will delete the tombstone at a similar time.
+
+Once the delay window passes, the RUV is _trimmed_. This moves the RUV minimum.
+
+We now need to consider the reason for this trimming process. Lets use these RUV's
+
+    Server A RUV                   Server B RUV
+    |-----|----------|----------|  |-----|----------|----------|
+    |     | s_uuid A | s_uuid B |  |     | s_uuid A | s_uuid B |
+    |-----|----------|----------|  |-----|----------|----------|
+    | min | T10      | T6       |  | min | T4       | T9       |
+    |-----|----------|----------|  |-----|----------|----------|
+    | max | T15      | T16      |  | max | T8       | T20      |
+    |-----|----------|----------|  |-----|----------|----------|
+
+The RUV for A on A does not overlap the range of the RUV for A on B (A min 10, B max 8).
+
+This means that a tombstone _could_ have been created at T9 _and then_ reaped. This would mean that
+B would not have perceived that delete and then the entry would become a zombie - back from the
+dead, risen again, escaping the grave, breaking the tombstone. This could have security consequences
+especially if the entry was a group providing access or a user who was needing to be deleted.
+
+To prevent this, we denote server B as _lagging_ since it is too old. We denote A as _advanced_
+since it has data newer that can not be applied to B.
+
+This will "freeze" B, where data will not be supplied to B, nor will data from B be accepted by
+other nodes. This is to prevent the risk of data corruption / zombies.
+
+There is some harm to extending the RUV trim / tombstone reaping window. This window could be
+expanded even to values as long as years. It would however increase the risk of conflicting changes
+however, where nodes that are segregated for extended periods have been accepting changes that may
+conflict with the other side of the topology.
```

Comment:
```suggestion
There is some harm to extending the RUV trim / tombstone reaping window. This window could be
expanded even to values as long as years. It would increase the risk of conflicting changes
however, where nodes that are segregated for extended periods have been accepting changes that may
conflict with the other side of the topology.
```